shell/browser/api/atom_api_web_contents.cc

```diff
@@ -1785,58 +1786,60 @@ void WebContents::Print(gin_helper::Arguments* args) {
   options.Get("footer", &footer);
 
   if (!(header.empty() && footer.empty())) {
-    settings.SetBoolean(printing::kSettingHeaderFooterEnabled, true);
+    settings.SetBoolKey(printing::kSettingHeaderFooterEnabled, true);
 
-    settings.SetString(printing::kSettingHeaderFooterTitle, header);
-    settings.SetString(printing::kSettingHeaderFooterURL, footer);
+    settings.SetStringKey(printing::kSettingHeaderFooterTitle, header);
+    settings.SetStringKey(printing::kSettingHeaderFooterURL, footer);
   } else {
-    settings.SetBoolean(printing::kSettingHeaderFooterEnabled, false);
+    settings.SetBoolKey(printing::kSettingHeaderFooterEnabled, false);
   }
 
   // We don't want to allow the user to enable these settings
   // but we need to set them or a CHECK is hit.
-  settings.SetInteger(printing::kSettingPrinterType, printing::kLocalPrinter);
-  settings.SetBoolean(printing::kSettingShouldPrintSelectionOnly, false);
-  settings.SetBoolean(printing::kSettingRasterizePdf, false);
-
-  // Set custom page ranges to print
-  std::vector<gin_helper::Dictionary> page_ranges;
-  if (options.Get("pageRanges", &page_ranges)) {
-    std::unique_ptr<base::ListValue> page_range_list(new base::ListValue());
-    for (auto& range : page_ranges) {
-      int from, to;
-      if (range.Get("from", &from) && range.Get("to", &to)) {
-        std::unique_ptr<base::DictionaryValue> range(
-            new base::DictionaryValue());
-        range->SetInteger(printing::kSettingPageRangeFrom, from);
-        range->SetInteger(printing::kSettingPageRangeTo, to);
-        page_range_list->Append(std::move(range));
-      } else {
-        continue;
-      }
-    }
-    if (page_range_list->GetSize() > 0)
-      settings.SetList(printing::kSettingPageRange, std::move(page_range_list));
-  }
+  settings.SetIntKey(printing::kSettingPrinterType, printing::kLocalPrinter);
+  settings.SetBoolKey(printing::kSettingShouldPrintSelectionOnly, false);
+  settings.SetBoolKey(printing::kSettingRasterizePdf, false);
+
+  // TODO(codebytere): figure out the value conversion for this
```

Comment:
i just wanted to get the thing compiling and mostly working - after this part is fixed this PR should be good to go